functests/vrf/vrf.go

```diff
@@ -0,0 +1,272 @@
+	}
+	apiclient := client.New("")
+
+	var nodesList []k8sv1.Node
+	var vrfBlue netattdefv1.NetworkAttachmentDefinition
+	var vrfRed netattdefv1.NetworkAttachmentDefinition
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+		nodesList, err = nodes.GetByRole(apiclient, "worker")
+		Expect(err).ToNot(HaveOccurred())
+		defIntName := getMainNodeInterface(apiclient, nodesList[0].Name)
+		Expect(err).ToNot(HaveOccurred())
+		vrfBlue = addVRFNad(apiclient, "test-vrf-blue", defIntName, VRFBlueName)
+		vrfRed = addVRFNad(apiclient, "test-vrf-red", defIntName, VRFRedName)
+	})
+	AfterEach(func() {
+		err := namespaces.CleanPods(TestNamespace, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("", func() {
+		// OCP-36305
+		DescribeTable("Integration: NAD, IPAM: static, Interfaces: 2, Scheme: 2 Pods 2 VRFs OCP Primary network overlap",
+			func(node string, ipStack string) {
+				var podClientNodeLabel string
+				var podServerNodeLabel string
+				var redVRFNetworkPrefix string
+				VRFParameters, err := newVRFTestParameters(node, ipStack)
+				Expect(err).ToNot(HaveOccurred())
+				if VRFParameters.Node == sameNode {
+					podClientNodeLabel = nodesList[0].Name
+					podServerNodeLabel = nodesList[0].Name
+					redVRFNetworkPrefix = "24"
+				} else {
+					if len(nodesList) < 2 {
+						Skip(fmt.Sprintf("There is not enough nodes to run test with following parameter %s", node))
+					}
+					podClientNodeLabel = nodesList[0].Name
+					podServerNodeLabel = nodesList[1].Name
+					redVRFNetworkPrefix = "8"
+				}
+				podClinetVRFRedOverlappingIP := getOverlapIP(apiclient, podClientNodeLabel)
+				podServerVRFRedOverlappingIP := getOverlapIP(apiclient, podServerNodeLabel)
+				if ipStack == ipStackIPv4 && net.ParseIP(podClinetVRFRedOverlappingIP).To4() == nil {
+					Skip("Skipping IPv4 test. Cluster supports IPv6 protocol")
+				} else if ipStack == ipStackIPv4 && net.ParseIP(podClinetVRFRedOverlappingIP).To4() != nil {
+					podClientVRFBlueIPAddress = "10.255.255.1"
+					podServerVRFBlueIPAddress = "10.255.255.2"
+				} else {
+					Skip("Unpsupported protocol parameter")
+				}
+				podClientIpamConfig := fmt.Sprintf(`[{"name": "%s", "mac": "%s", "ips": ["%s/24"]}, {"name": "%s", "mac": "%s", "ips": ["%s/%s"]}]`,
+					vrfBlue.Name, "20:04:0f:f1:88:01", podClientVRFBlueIPAddress, vrfRed.Name, "20:04:0f:f1:88:02", podClinetVRFRedOverlappingIP, redVRFNetworkPrefix)
+				PodClient := redefineAsPrivileged(
+					pods.RedefinePodWithNetwork(pods.DefinePodOnNode(TestNamespace, podClientNodeLabel), podClientIpamConfig))
+				podServerIpamConfig := fmt.Sprintf(`[{"name": "%s", "mac": "%s", "ips": ["%s/24"]}, {"name": "%s", "mac": "%s", "ips": ["%s/%s"]}]`,
+					vrfBlue.Name, "20:04:0f:f1:88:03", podServerVRFBlueIPAddress, vrfRed.Name, "20:04:0f:f1:88:04", podServerVRFRedOverlappingIP, redVRFNetworkPrefix)
+				PodServer := redefineAsPrivileged(
```

Comment:
Can you give a name to the pods (using generatename). This will make debugging easier.